### DISCUSSION TITLE
Component nesting

### DISCUSSION THREAD

**Discussion by PGorshkov** [cssinjs/jss#1542] [Q&A]
Hey! I am implementing a component library for VueJS2 and there is a need to keep each style separately for the component.
But there was a problem with nesting and the impact on styles of nested styles.
Once created, the StyleSheet is not initialized internally to a reference to another class.
How can I work around this problem?
Style component Button:
`
rButton: {
    cursor: 'pointer',
    border: 0,
    display: 'flex',
    '& $rIcon': {
      marginLeft: 20
    }
  }
`
Style component Icon:
`
rIcon: {
    display: 'inline-flex'
  }
  `
 After the button is initialized, the icon class after addRule is not rewritten in styles to the icon class

**Comment by fardad-dev** [comment]
https://github.com/cssinjs/jss/discussions/1491
